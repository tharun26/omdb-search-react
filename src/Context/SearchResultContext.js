import React, { useState, createContext } from 'react';
import { fetchSearchData } from '../Services/SearchService';

export const SearchResultContext = createContext();

export const SearchResultContextProvider = ({ children, values }) => {
  const [movies, setMovies] = useState([]);
  const [totalResults, setTotalresults] = useState(0);
  const [hasMore, setHasMore] = useState(true);
  const [searchTerm, setSearchTerm] = useState('');
  const [selectedYear, setSelectedYear] = useState('');
  const [selectedType, setSelectedType] = useState('');
  const [loading, setLoading] = useState(false);

  const [showModal, setShowModal] = useState(false);
  const [movieID, setMovieID] = useState('');
  const handleOnClick = (id) => {
    setShowModal(true);
    setMovieID(id);
  };

  const fetchData = (page) => {
    setLoading(true);
    if (page > 1 && page > Math.ceil(totalResults / 10)) {
      setHasMore(false);
      setLoading(false);
      return;
    }
    const searchResult = fetchSearchData(
      searchTerm,
      selectedYear,
      selectedType,
      page
    );
    searchResult.then((result) => {
      if (movies.length >= result['totalResult']) {
        setHasMore(false);
        setLoading(false);
        return;
      }
      if (result) {
        const moviesReturned = result['movies'];
        const filterdMovies = moviesReturned.filter(
          (item) => item.movie.poster !== 'N/A'
        );
        const newMovies = filterdMovies.map((item) => item.movie);
        // const newMovies = moviesReturned.map((item) => item.movie);
        setMovies((prevMovies) => {
          return [...prevMovies, ...newMovies];
        });
        result['totalResult'] && setTotalresults(result['totalResult']);
        setLoading(false);
      }
    });
  };

  return (
    <SearchResultContext.Provider
      value={{
        movies,
        setMovies,
        searchTerm,
        setSearchTerm,
        setSelectedYear,
        setSelectedType,
        totalResults,
        setTotalresults,
        hasMore,
        setHasMore,
        loading,
        setLoading,
        fetchData,
        showModal,
        setShowModal,
        movieID,
        setMovieID,
        handleOnClick,
        ...values,
      }}
    >
      {children}
    </SearchResultContext.Provider>
  );
};

export default SearchResultContext;
