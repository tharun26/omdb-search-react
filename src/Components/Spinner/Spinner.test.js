import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });
import { SearchResultContextProvider } from '../../Context/SearchResultContext';

import Spinner from './Spinner';

describe('Spinner', () => {
  it('should render Spinner', async () => {
    jest.useRealTimers();
    const component = mount(<Spinner></Spinner>);
    expect(component).toMatchSnapshot();
  });
});
