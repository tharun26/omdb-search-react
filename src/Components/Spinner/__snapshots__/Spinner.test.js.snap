// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Spinner 1`] = `
<Spinner>
  <div
    className="spinner-overlay"
  >
    <div
      className="spinner-container"
    />
  </div>
</Spinner>
`;
